xdsl/interpreters/pdl_interp.py

```diff
@@ -291,38 +292,47 @@ def run_create_operation(
 
         # Split args into operands, attributes and result types based on operand segments
         operands = list(args[0 : len(op.input_operands)])
-        attributes = list(
+
+        attr_names: list[str] = [
+            cast(StringAttr, name).data for name in op.input_attribute_names.data
+        ]
+
+        assert issubclass(op_type, IRDLOperation)
+        existing_properties = op_type.get_irdl_definition().properties.keys()
+
+        attributes: dict[str, Attribute] = {}
+        properties: dict[str, Attribute] = {}
+        for name, prop_or_attr in zip(
+            attr_names,
             args[
                 len(op.input_operands) : len(op.input_operands)
                 + len(op.input_attributes)
-            ]
-        )
+            ],
+        ):
+            if name in existing_properties:
+                properties[name] = prop_or_attr
+            else:
+                attributes[name] = prop_or_attr
         result_types = list(args[len(op.input_operands) + len(op.input_attributes) :])
 
         # Verify all arguments have correct types
         for operand in operands:
             assert isinstance(operand, SSAValue)
-        for attr in attributes:
+        for attr in attributes.values():
             assert isinstance(attr, Attribute)
```

Comment:
Do we need to assert this?

Reply:
I guess not. I've removed all those type assertions